src/TFMiniPlus.cpp

```diff
@@ -11,72 +11,70 @@
  */
 
 #include "TFMiniPlus.h"
+#include "TFMiniPlusConstants.h"
 
-#define DATA_FRAME_MARKER 0x59
-#define DATA_FRAME_LENGTH 9
-#define CMD_FRAME_MARKER 0x5A
-#define MAX_CMD_RESPONSE_LENGTH 8
+template <std::size_t N>
+constexpr std::size_t buffer_size(const uint8_t (&buffer)[N]) noexcept {
+  return N;
+}
 
 void TFMiniPlus::begin(Stream* serial) { _stream = serial; }
 
 bool TFMiniPlus::readData() {
-  uint8_t buffer[9];
+  uint8_t buffer[DATA_FRAME_LENGTH];
 
-  TFMiniPlus::resetBuffer(_readDataBuffer, 9);
+  TFMiniPlus::resetBuffer(_readDataBuffer, buffer_size(_readDataBuffer));
 
   // skip to the first data frame header
   TFMiniPlus::skipToFrameHeader(DATA_FRAME_MARKER);
 
   // get the data body "all - frame marker byte"
-  _stream->readBytes(buffer, DATA_FRAME_LENGTH - 1);
+  _stream->readBytes(&buffer[1], DATA_FRAME_LENGTH - 1);
```

Comment:
Neat .. so instead of giving it the buffer array i just give it a pointer to the first element i want to overwrite and it just fills in the blanks ?

Reply:
Yes. In C/C++ array variables are functionally the same as pointers. Wherever an array is expected you can pass a pointer